src/main/resources/com/tresys/nitf/xsd/nitf.dfdl.xsd

```diff
@@ -165,36 +182,58 @@ The message root is 'NITF'.
+                <xs:annotation>
+                  <xs:appinfo source="http://www.ogf.org/dfdl/">
+                    <dfdl:setVariable ref="nitf:isNITF20" value="{ if (fn:exists(../ClassificationInfo2.0)) then 'true' else 'false' }" />
+                  </xs:appinfo>
+                </xs:annotation>
+              </xs:element>
               <xs:element name="FileNumberOfCopies" type="nitf:BCS-NP" dfdl:length="5" />
               <xs:element name="Encryption" type="nitf:BCS-N" dfdl:length="1" />
-              <xs:element name="FileBackgroundColor">
-                <xs:complexType>
-                  <xs:sequence>
-                    <xs:element name="Red" type="xs:hexBinary" dfdl:lengthKind="explicit" dfdl:length="1" />
-                    <xs:element name="Green" type="xs:hexBinary" dfdl:lengthKind="explicit" dfdl:length="1" />
-                    <xs:element name="Blue" type="xs:hexBinary" dfdl:lengthKind="explicit" dfdl:length="1" />
-                  </xs:sequence>
-                </xs:complexType>
-              </xs:element>
-              <xs:element name="OriginatorsName" type="nitf:ECS-A" dfdl:length="24" nillable="true" dfdl:ref="nitf:nilString" />
-              <xs:element name="OriginatorsPhoneNumber" type="nitf:ECS-A" dfdl:length="18" nillable="true" dfdl:ref="nitf:nilString" />
+              <xs:choice dfdl:choiceDispatchKey="{ $nitf:isNITF20 }">
+                <xs:element name="HeaderInfo2.1" dfdl:choiceBranchKey="false">
+                  <xs:complexType>
+                    <xs:sequence>
+                      <xs:element name="FileBackgroundColor">
+                        <xs:complexType>
+                          <xs:sequence>
+                            <xs:element name="Red" type="xs:hexBinary" dfdl:lengthKind="explicit" dfdl:length="1" />
+                            <xs:element name="Green" type="xs:hexBinary" dfdl:lengthKind="explicit" dfdl:length="1" />
+                            <xs:element name="Blue" type="xs:hexBinary" dfdl:lengthKind="explicit" dfdl:length="1" />
+                          </xs:sequence>
+                        </xs:complexType>
+                      </xs:element>
+                      <xs:element name="OriginatorsName" type="nitf:ECS-A" dfdl:length="24" nillable="true" dfdl:ref="nitf:nilString" />
+                      <xs:element name="OriginatorsPhoneNumber" type="nitf:ECS-A" dfdl:length="18" nillable="true" dfdl:ref="nitf:nilString" />
+                    </xs:sequence>
+                  </xs:complexType>
+                </xs:element>
+                <xs:element name="HeaderInfo2.0" dfdl:choiceBranchKey="true">
+                  <xs:complexType>
+                    <xs:sequence>
+                      <xs:element name="OriginatorsName" type="nitf:ECS-A" dfdl:length="27" nillable="true" dfdl:ref="nitf:nilString" />
+                      <xs:element name="OriginatorsPhoneNumber" type="nitf:ECS-A" dfdl:length="18" nillable="true" dfdl:ref="nitf:nilString" />
```

Comment:
Looks like OriginatorsPhoneNumer can e moved out of the choice?

Also, I wonder if there's a benefit to avoiding the HeaderInfo elements? FileBackGroundColor could be made optional with occursCount depending on the nitif version. And OriginatosrsName could have a length expression depending on the version as well. Might make the infoset a little cleaner. I prefer to avoid version specific elements when possible. That way if someone ever needs to access a field (e.g. OriginatorsName) they don't need to know the version to figure out which HeaderInfo to look into.